group_project_v2/mixins.py

```diff
@@ -266,18 +429,39 @@ class DashboardRootXBlockMixin(ProjectAPIXBlockMixin):
     TARGET_STUDENTS = 'target_students'
     TARGET_WORKGROUPS = 'target_workgroups'
 
-    def _append_context_parameters_if_not_present(self, context):
+    FILTERED_STUDENTS = "filtered_students"
+
+    def _add_students_and_workgroups_to_context(self, context, filter_by_organisation_id=None):
         """
-        Appends target students and target workgroups parameters, if not already present in context
         :param dict context: XBlock view context
+        :param int filter_by_organisation_id:
+            If not None students not belonging to organisation represented by
+            given id will be filtered out from dashboard view.
+
         :rtype: None
         """
-        workgroups, users = self.get_workgroups_and_students()
-        if self.TARGET_STUDENTS not in context:
-            context[self.TARGET_STUDENTS] = users
+        workgroups, students = self.get_workgroups_and_students()
+
+        context[self.TARGET_STUDENTS] = list(students)
+        context[self.TARGET_WORKGROUPS] = list(workgroups)
+        context[self.FILTERED_STUDENTS] = set()
+
+        if filter_by_organisation_id is None:
+            filter_by_organisation_id = tuple()
```

Comment:
@jbzdak see comment about same meaning of `None` and empty collection in `get_organisation_filter_for_user`